resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -1105,8 +1106,19 @@ static boolean doResolveAllCached(String hostname,
                                       DnsRecord[] additionals,
                                       Promise<List<InetAddress>> promise,
                                       DnsCache resolveCache,
+                                      String[] searchDomains,
+                                      int ndots,
                                       InternetProtocolFamily[] resolvedInternetProtocolFamilies) {
-        final List<? extends DnsCacheEntry> cachedEntries = resolveCache.get(hostname, additionals);
+        List<? extends DnsCacheEntry> cachedEntries = resolveCache.get(hostname, additionals);
+        if (searchDomains != null && ndots != 0 && !StringUtil.endsWith(hostname, '.')) {
+            for (String searchDomain : searchDomains) {
+                if (cachedEntries != null && !cachedEntries.isEmpty()) {
+                    break;
+                }
+                final String initialHostname = hostname + '.' + searchDomain;
+                cachedEntries = resolveCache.get(initialHostname, additionals);
+            }
+        }
```

Comment:
@qnnn I think we should do this as a separate PR as its a different problem / bug. Can you open an extra PR with just this change and a unit test ?